blocks/variables_dynamic.ts

```diff
@@ -8,35 +8,29 @@
  * @fileoverview Variable blocks for Blockly.
  * @suppress {checkTypes}
  */
-'use strict';
 
-goog.module('Blockly.libraryBlocks.variablesDynamic');
+import * as goog from '../closure/goog/goog.js';
+goog.declareModuleId('Blockly.libraryBlocks.variablesDynamic');
 
-/* eslint-disable-next-line no-unused-vars */
-const AbstractEvent = goog.requireType('Blockly.Events.Abstract');
-const ContextMenu = goog.require('Blockly.ContextMenu');
-const Extensions = goog.require('Blockly.Extensions');
-const Variables = goog.require('Blockly.Variables');
-const xml = goog.require('Blockly.utils.xml');
-/* eslint-disable-next-line no-unused-vars */
-const {Block} = goog.requireType('Blockly.Block');
-// const {BlockDefinition} = goog.requireType('Blockly.blocks');
-// TODO (6248): Properly import the BlockDefinition type.
-/* eslint-disable-next-line no-unused-vars */
-const BlockDefinition = Object;
-const {Msg} = goog.require('Blockly.Msg');
-const {createBlockDefinitionsFromJsonArray, defineBlocks} = goog.require('Blockly.common');
+import {Abstract as AbstractEvent} from '../core/events/events_abstract.js';
+import type {ContextMenuOption, LegacyContextMenuOption} from '../core/contextmenu_registry.js';
+import * as ContextMenu from '../core/contextmenu.js';
+import {FieldVariable} from '../core/field_variable.js';
+import * as Extensions from '../core/extensions.js';
+import * as Variables from '../core/variables.js';
+import type {WorkspaceSvg} from '../core/workspace_svg.js';
+import * as xml from '../core/utils/xml.js';
+import type {Block} from '../core/block.js';
+import {Msg} from '../core/msg.js';
+import {createBlockDefinitionsFromJsonArray, defineBlocks} from '../core/common.js';
```

Comment:
I did it that way because lowercase letters come after uppercase in asciibetical order—but now I realise I'm not actually sure what the rules are (and they no longer exist in the styleguide, since it just says to use the tooling).  Maybe we should be ordering by filename?  Who knows.  Anyway… feel free to do something different if you have a principled reason for doing so.